core/system.cpp

```diff
@@ -282,7 +285,54 @@ void Device::device_thread(Device *self)
     }
 }
 
-void Device::send_heartbeat(Device &self)
+std::string System::name(uint8_t component_id)
+{
+    switch (component_id) {
+        case MAV_COMP_ID_AUTOPILOT1:
+            return "Autopilot";
+        case MAV_COMP_ID_CAMERA...MAV_COMP_ID_CAMERA6:
+            return "Camera";
+        case MAV_COMP_ID_GIMBAL:
+            return "Gimbal";
+        default:
+            return "Unsupported component";
+    }
+}
+
+void System::add_new_component(uint8_t component_id)
+{
+    auto res_pair = _components.insert(component_id);
+    if (res_pair.second) {
+        LogDebug() << "Component " << name(component_id) << " added.";
+    }
+}
+
+size_t System::total_components() const
+{
+    return _components.size();
+}
+
+bool System::is_standalone() const
```

Comment:
HOw is this supposed to be used? i.e. what it for/what does it mean? 

I guess I'm just concerned that an autopilot on its own is technically also "standalone".

Reply:
I think, if a system has autopilot, its subjected to move; isn't it ?